File: library/src/main/java/me/proxer/library/api/media/CalendarEndpoint.java
package me.proxer.library.api.media;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;
import me.proxer.library.entity.media.CalendarEntry;

import java.util.List;

/**
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public final class CalendarEndpoint implements Endpoint<List<CalendarEntry>> {

    private final InternalApi internalApi;

    CalendarEndpoint(final InternalApi internalApi) {
        this.internalApi = internalApi;
    }

    @Override
    public ProxerCall<List<CalendarEntry>> build() {
        return internalApi.calendar();
    }
}

File: library/src/main/kotlin/me/proxer/library/api/media/CalendarEndpoint.kt
package me.proxer.library.api.media

import me.proxer.library.api.Endpoint
import me.proxer.library.api.ProxerCall
import me.proxer.library.entity.media.CalendarEntry

/**
 * Endpoint for retrieving the anime calendar/schedule.
 *
 * @author Ruben Gees
 */
class CalendarEndpoint internal constructor(private val internalApi: InternalApi) : Endpoint<List<CalendarEntry>> {

    override fun build(): ProxerCall<List<CalendarEntry>> {
        return internalApi.calendar()
    }
}
